apis/core/v1alpha2/readinessprovider_types.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ReadinessProviderState string
+
+const (
+	ProviderSuccessState    = ReadinessProviderState("success")
+	ProviderFailureState    = ReadinessProviderState("failure")
+	ProviderInProgressState = ReadinessProviderState("inprogress")
+)
+
+type ReadinessConditionState string
+
+const (
+	ConditionSuccessState    = ReadinessConditionState("success")
+	ConditionFailureState    = ReadinessConditionState("failure")
+	ConditionInProgressState = ReadinessConditionState("inprogress")
+)
+
+// ReadinessProviderSpec defines the desired state of ReadinessProvider
+type ReadinessProviderSpec struct {
+	// CheckRef is the name of the check that the current provider satisfies
+	CheckRef string `json:"checkRef"`
+
+	// Repeatable denotes if the provider should be re-evaluated periodically
+	//+kubebuilder:validation:Optional
+	Repeatable bool `json:"repeatable"`
+
+	// RepeatInterval is the re-evaluation interval
```

Comment:
I'm thinking, can we just use `RepeatInterval *metav1.Time` and `Repeatable bool` is not needed. Since its a pointer ref, if the value is nil => no periodic evaluation needed. Any specified value will enforce the periodic evaluation.

Reply:
Done